packages/boxel-ui/addon/src/helpers/color-tools.ts

```diff
@@ -66,22 +105,221 @@ export function hslToRgb(h: number, s: number, l: number): RGB {
+  return `rgb(${Math.round(r)}, ${Math.round(g)}, ${Math.round(b)})`;
+}
+
+// Format RGBA as rgba() string (preserves alpha)
+export function rgbaToRgbaString({ r, g, b, a }: RGBA): string {
+  return `rgba(${Math.round(r)}, ${Math.round(g)}, ${Math.round(b)}, ${a.toFixed(2)})`;
+}
+
+// Format RGBA as HSL string (returns hsl/hsla depending on alpha)
+export function rgbaToHslString(rgba: RGBA): string {
+  const { h, s, l } = rgbaToHsl(rgba);
+  return `hsl(${Math.round(h)}, ${Math.round(s)}%, ${Math.round(l)}%)`;
+}
+
+// Format RGBA as HSLA string (includes alpha)
+export function rgbaToHslaString(rgba: RGBA): string {
+  const { h, s, l } = rgbaToHsl(rgba);
+  const { a } = rgba;
+  return `hsla(${Math.round(h)}, ${Math.round(s)}%, ${Math.round(l)}%, ${a.toFixed(2)})`;
+}
+
+// Format RGBA as HSB string (default minus alpha support)
+export function rgbaToHsbString(rgba: RGBA): string {
+  const { h, s, v } = rgbaToHsv(rgba);
+  return `hsb(${Math.round(h)}, ${Math.round(s)}%, ${Math.round(v)}%)`;
+}
+
+// -------------------- Format helpers --------------------
+// Return string representation for the requested format
+export function rgbaToFormatString(
+  rgba: RGBA,
+  format: RichColorFormat,
+): string {
+  switch (format) {
+    case 'hex':
+      return rgbaToHexString(rgba);
+    case 'rgb':
+      return rgba.a < 1 ? rgbaToRgbaString(rgba) : rgbaToRgbString(rgba);
+    case 'hsl':
+      return rgba.a < 1 ? rgbaToHslaString(rgba) : rgbaToHslString(rgba);
+    case 'hsb':
+      return rgbaToHsbString(rgba);
+    case 'css':
+      return rgbaToRgbaString(rgba);
+    default:
+      return rgbaToHexString(rgba);
+  }
+}
+
+// Guess the color format from an arbitrary string
+export function detectColorFormat(input: string): RichColorFormat {
+  const trimmed = input.trim();
+  if (/^#?[a-f\d]{3}([a-f\d]{3})?([a-f\d]{2})?$/i.test(trimmed)) {
+    return 'hex';
+  }
+  if (/^rgb\(/i.test(trimmed)) {
+    return 'rgb';
+  }
+  if (/^hsl\(/i.test(trimmed)) {
+    return 'hsl';
+  }
+  if (/^hsb\(/i.test(trimmed)) {
+    return 'hsb';
+  }
+  return 'css';
+}
+
+// -------------------- Contrast helpers --------------------
+// Calculate relative luminance from an RGB triple
 export function calculateLuminance({ r, g, b }: RGB): number {
   const [red, green, blue] = [r, g, b].map((c) => {
-    // divide by 255 to get each color's value between 0 and 1
     const val = c / 255;
-    // apply gamma correction
     return val <= 0.03928 ? val / 12.92 : Math.pow((val + 0.055) / 1.055, 2.4);
   }) as [number, number, number];
 
-  // formula for relative luminance: L = 0.2126 * R + 0.7152 * G + 0.0722 * B
   return 0.2126 * red + 0.7152 * green + 0.0722 * blue;
 }
 
 // takes in luminance values to calculate contrast ratio
 export function calculateContrast(lum1: number, lum2: number): number {
-  // formula for contrast ratio: (L1 + 0.05) / (L2 + 0.05) where L1 is the lighter color
```

Comment:
these comments are helpful for understanding what's going on, so it would be good to keep them